google-ads/src/test/java/com/google/ads/googleads/lib/utils/FieldMasksTest.java

```diff
@@ -1,24 +1,101 @@
 package com.google.ads.googleads.lib.utils;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
 
-import com.google.ads.googleads.v5.resources.Campaign;
+import com.google.ads.googleads.v6.enums.CampaignStatusEnum.CampaignStatus;
+import com.google.ads.googleads.v6.resources.Campaign;
 import com.google.common.collect.ImmutableList;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 @RunWith(JUnit4.class)
 public class FieldMasksTest {
+  private Campaign campaign;
+
+  @Before
+  public void setUp() {
+    Campaign.Builder builder =
+        Campaign.newBuilder().setName("New campaign").setStatus(CampaignStatus.PAUSED);
+    builder.getManualCpcBuilder().setEnhancedCpcEnabled(false);
+    campaign = builder.build();
+  }
 
   // Regression test for https://github.com/googleads/google-ads-java/issues/344.
   @Test
   public void optionalField_withDefaultValue_detectsChange() {
     Campaign.Builder builder = Campaign.newBuilder();
     builder.getManualCpcBuilder().setEnhancedCpcEnabled(false);
-    Campaign campaign = builder.build();
+    campaign = builder.build();
     assertEquals(
         ImmutableList.of("manual_cpc.enhanced_cpc_enabled"),
         FieldMasks.allSetFieldsOf(campaign).getPathsList());
   }
+
+  @Test
+  public void getFieldValue_gets_basic_field() {
+    assertEquals(FieldMasks.getFieldValue("name", campaign), campaign.getName());
+  }
+
+  @Test
+  public void getFieldValue_gets_nested_field() {
+    assertEquals(
+        FieldMasks.getFieldValue("manual_cpc.enhanced_cpc_enabled", campaign),
+        campaign.getManualCpc().getEnhancedCpcEnabled());
+  }
+
+  @Test
+  public void getFieldValue_gets_top_nested_field() {
+    assertEquals(FieldMasks.getFieldValue("manual_cpc", campaign), campaign.getManualCpc());
+  }
+
+  @Test
+  public void getFieldValue_gets_enum_field() {
+    assertEquals(
```

Comment:
I noticed none of the tests attempts to assign the result of `FieldMasks.getFieldValue()` to a specific type (e.g., `CampaignStatusEnum.CampaignStatus` here). Could you add that to make sure the casts work as expected?

Reply:
I've added tests here to demonstrate all cases of assignment. The only "quirk" is that you cannot assign an enum value directly. You have to assign by number because the value returned is an `EnumValueDescriptor`. Even with the casting change you mentioned in your other comment, which I've made, you still have to approach it in this way.

Reply:
Thanks, and I see now what you mean about enums. Perhaps a future enhancement would be to return the value as the typed enum, but there are edge cases we should consider first (like unknown values), so I won't hold up this enhancement for that.